README.md

```diff
@@ -2,21 +2,177 @@
+
+#### ADMIN API Server
+```mermaid
+   graph LR;
+   CLIENT{Client}
+   CLIENT --> ADMIN_API
+   CONFIG{{Config file}}
+   MAIN --1.reads--> CONFIG
+   subgraph "Casper sidecar"
+      MAIN[main.rs]
+      MAIN --2.spawns--> ADMIN_API
+      ADMIN_API["ADMIN API"]
+   end
+```
+
+Sidecar offers an ADMIN API that allows the operator to look into the current operation of Sidecar. The ADMIN API is optional. For details please see the [admin server configuration](#admin-server).
+
+#### RPC API Server
+```mermaid
+   graph LR;
+   CLIENT{Client}
+   CLIENT --> RPC_API
+   CONFIG{{Config file}}
+   MAIN --1.reads--> CONFIG
+   CASPER_NODE(("Casper Node binary port"))
+   RPC_API --forwards request--> CASPER_NODE
+   subgraph "Casper sidecar"
+      MAIN[main.rs]
+      MAIN --2.spawns--> RPC_API
+      RPC_API["RPC JSON API"]
+   end
+```
+Sidecar offers an RPC JSON API. This module is optional. It is a json bridge between the end users and the binary port od casper node. The RPC API server forwards the requests to the Casper Node binary port. 
+For more details on how the RPC JSON API works please see the [rpc sidecar README](rpc_sidecar/README.md).
+
+To discuss the configuration of RPC API let's consider the following configuration section:
+
+```
+[rpc_server.main_server]
+enable_server = true
+address = '0.0.0.0:7777'
+qps_limit = 100
+max_body_bytes = 2_621_440
+cors_origin = ''
+
+[rpc_server.node_client]
+address = '127.0.0.1:28101'
+max_request_size_bytes = 4_194_304
+max_response_size_bytes = 4_194_304
+request_limit = 3
+request_buffer_size = 16
+
+[rpc_server.speculative_exec_server]
+enable_server = true
+address = '0.0.0.0:7778'
+qps_limit = 1
+max_body_bytes = 2_621_440
+cors_origin = ''
+
+
+[rpc_server.node_client.exponential_backoff]
+initial_delay_ms = 1000
+max_delay_ms = 32_000
+coefficient = 2
+max_attempts = 30
+```
+
+* `main_server.enable_server` - If set to true, the RPC API server will be enabled
+* `main_server.address` - Address under which the main RPC API server will be available
+* `main_server.qps_limit` - The maximum number of requests per second
+* `main_server.max_body_bytes` - Maximum body size of request to API in bytes
+* `main_server.cors_origin` - allow configuration of the CORS Origin.
+
+* `speculative_exec_server.enable_server` - If set to true, the speculative RPC API server will be enabled
+* `speculative_exec_server.address` - Address under which the speculative RPC API server will be available
+* `speculative_exec_server.qps_limit` - The maximum number of requests per second
+* `speculative_exec_server.max_body_bytes` - Maximum body size of request to API in bytes
+* `speculative_exec_server.cors_origin` - allow configuration of the CORS Origin.
+
+* `node_client.address` - Address of the Casper Node binary port
```

Comment:
Please also replace "binary port" with one of these ports referenced in the docs. Same comment for all other occurrences: 
[Networking port: 35000](https://docs.casper.network/operators/setup/node-endpoints/#35000)
[JSON-RPC HTTP server port: 7777](https://docs.casper.network/operators/setup/node-endpoints/#7777)
[REST HTTP server port: 8888](https://docs.casper.network/operators/setup/node-endpoints/#8888)
[SSE HTTP event stream server port: 9999](https://docs.casper.network/operators/setup/node-endpoints/#9999)

```suggestion
* `node_client.address` - Address of a Casper node binary port.
```

Reply:
Binary port is a new port that the node will expose since 2.0. I don't know what the deault value for it is, it's not in the docs for sure.